### PR TITLE
fix: map headless auth pw recovery to auto-verified attrs

### PR DESCRIPTION
**Pull request by edwardfoyle**
#### Description of changes
passwordRecovery options were mapped to autoVerified attributes in the v1 headless schema. the v2 schema instead specified autoVerifiedAttributes explicitly. Additionally, a translation function is added to map incoming v1 requests to the correct v2 request.

Related docs PR: https://github.com/aws-amplify/docs/pull/4046

#### Issue #, if available
fixes https://github.com/aws-amplify/amplify-cli/issues/9756

#### Description of how you validated changes
Manually, added unit tests, updated e2e tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by danielleadams — APPROVED** [review]

File: packages/amplify-headless-interface/schemas/auth/1/AddAuthRequest.schema.json [hunk 1]
```diff
@@ -139,7 +139,7 @@
                     ]
                 },
                 "passwordRecovery": {
-                    "description": "If defined, specifies password recovery configiuration. Default is email recovery.",
+                    "description": "DEPRECATED. Use autoVerifiedAttributes in headless schema version 2 intead. If defined, specifies password recovery configiuration. Default is email recovery.",
```
Comment (yppartyk) [inline packages/amplify-headless-interface/schemas/auth/1/AddAuthRequest.schema.json:142]: Typo: "instead"

File: packages/amplify-headless-interface/schemas/auth/1/UpdateAuthRequest.schema.json [hunk 2]
```diff
@@ -99,7 +99,7 @@
                     "description": "If defined, specifies password constraint configuration. Default is minimum length of 8 characters."
                 },
                 "passwordRecovery": {
-                    "description": "If defined, specifies password recovery configiuration. Default is email recovery.",
+                    "description": "DEPRECATED. Use autoVerifiedAttributes in headless schema version 2 intead. If defined, specifies password recovery configiuration. Default is email recovery.",
```
Comment (yppartyk) [inline packages/amplify-headless-interface/schemas/auth/1/UpdateAuthRequest.schema.json:102]: Typo: "instead"

File: packages/amplify-headless-interface/schemas/auth/1/AddAuthRequest.schema.json (hunk 1 above)
Comment (edwardfoyle) [inline packages/amplify-headless-interface/schemas/auth/1/AddAuthRequest.schema.json:142]:
```suggestion
                    "description": "DEPRECATED. Use autoVerifiedAttributes in headless schema version 2 instead. If defined, specifies password recovery configuration. Default is email recovery.",
```

File: packages/amplify-headless-interface/schemas/auth/1/UpdateAuthRequest.schema.json (hunk 2 above)
Comment (edwardfoyle) [inline packages/amplify-headless-interface/schemas/auth/1/UpdateAuthRequest.schema.json:102]:
```suggestion
                    "description": "DEPRECATED. Use autoVerifiedAttributes in headless schema version 2 instead. If defined, specifies password recovery configuration. Default is email recovery.",
```

**Review by yppartyk — APPROVED** [review]

**Pull request merged**